services/cron/tasks_extended.go

```diff
@@ -160,7 +160,9 @@ func initExtendedTasks() {
 	registerGarbageCollectRepositories()
 	registerRewriteAllPublicKeys()
 	registerRewriteAllPrincipalKeys()
-	registerRepositoryUpdateHook()
+	if !setting.DisableGitHooks {
```

Comment:
No, I don't think this is right. `DisableGitHooks` should be `DisableExtraGitHooks`. We need to update Gitea internal git hooks.